## Homework 1
### Tiffany Tang
### tt2405

#### 1. First uncompress the file using "gunzip weight-height.csv.gz".

**I used the unzipped file provided to us**

#### 2. Import data from this file into the data frame "wh"

- wh <- read.csv("weight-height.csv",header=TRUE)

- You can read the compressed file directly, but you sometimes need to check contents after importing, so best to have an uncompressed version at hand to use with excel to check the import

In [1]:
wh <- read.csv("weight-height.csv", header = TRUE)

#### 3. How many males and females are in this file ? (hint: use the command summary(wh))

In [2]:
summary(wh) #shows us the columns
#know that total amount of people is 10,000, so male and female must total this

    Gender              Height          Weight     
 Length:10000       Min.   :54.26   Min.   : 64.7  
 Class :character   1st Qu.:63.51   1st Qu.:135.8  
 Mode  :character   Median :66.32   Median :161.2  
                    Mean   :66.37   Mean   :161.4  
                    3rd Qu.:69.17   3rd Qu.:187.2  
                    Max.   :79.00   Max.   :270.0  

In [4]:
head(wh) #just to get a look at the data and the pattern I need for grep

Gender Height   Weight  
1 Male   73.84702 241.8936
2 Male   68.78190 162.3105
3 Male   74.11011 212.7409
4 Male   71.73098 220.0425
5 Male   69.88180 206.3498
6 Male   67.25302 152.2122

In [15]:
#I am going to use grep to find the amounts of matches for male and female
num_males <- length(grep("Male", wh$Gender)) #use grep to find match for 'Male" in column 'Gender'
num_females <- length(grep("Female", wh$Gender)) #use grep to find match for 'Female' in column 'Gender'
cat("Number of males:", num_males, "\n")
cat("Number of females:", num_females, "\n")

Number of males: 5000 
Number of females: 5000 


**There are 5000 males and 5000 females**

#### 4. Find the mean and standard deviation in height and weight of the whole group and the males and females separately.

In [21]:
#whole group
#this information is also provided in the summary
mean_height <- mean(wh$Height)
mean_weight <- mean(wh$Weight)

sd_height <- sd(wh$Height)
sd_weight <- sd(wh$Weight)

cat("Mean Height:", round(mean_height, 2), "\n")
cat("Standard Deviation Height:", round(sd_height, 2), "\n")
cat("Mean Weight:", round(mean_weight, 2), "\n")
cat("Standard Deviation Weight:", round(sd_weight, 2), "\n")

Mean Height: 66.37 
Standard Deviation Height: 3.85 
Mean Weight: 161.44 
Standard Deviation Weight: 32.11 


In [28]:
#Just the males:
wh_males <- wh[wh$Gender == "Male", ]#this creates a dataframe of just males
#create a dataframe from rows in the column that match Male under Gender, then get all columns

mean_height_male <- mean(wh_males$Height)
mean_weight_male <- mean(wh_males$Weight)

sd_height_male <- sd(wh_males$Height)
sd_weight_male <- sd(wh_males$Weight)

cat("Mean Height Males:", round(mean_height_male, 2), "\n")
cat("Standard Deviation Height Males:", round(sd_height_male, 2), "\n")
cat("Mean Weight Males:", round(mean_weight_male, 2), "\n")
cat("Standard Deviation Weight Males:", round(sd_weight_male, 2), "\n")

Mean Height Males: 69.03 
Standard Deviation Height Males: 2.86 
Mean Weight Males: 187.02 
Standard Deviation Weight Males: 19.78 


In [29]:
#Just the Females
#following same procedure as I did for males
wh_females <- wh[wh$Gender == "Female", ]#Same method as above for females

mean_height_female <- mean(wh_females$Height)
mean_weight_female <- mean(wh_females$Weight)

sd_height_female <- sd(wh_females$Height)
sd_weight_female <- sd(wh_females$Weight)

cat("Mean Height Females:", round(mean_height_female, 2), "\n")
cat("Standard Deviation Height Females:", round(sd_height_female, 2), "\n")
cat("Mean Weight Females:", round(mean_weight_female, 2), "\n")
cat("Standard Deviation Weight Females:", round(sd_weight_female, 2), "\n")

Mean Height Females: 63.71 
Standard Deviation Height Females: 2.7 
Mean Weight Females: 135.86 
Standard Deviation Weight Females: 19.02 


**For the whole group:**
- Mean Height: 66.37 
- Standard Deviation Height: 3.85 
- Mean Weight: 161.44 
- Standard Deviation Weight: 32.11

**For Males**
- Mean Height Males: 69.03 
- Standard Deviation Height Males: 2.86 
- Mean Weight Males: 187.02 
- Standard Deviation Weight Males: 19.78

**For Females**
- Mean Height Females: 63.71 
- Standard Deviation Height Females: 2.7 
- Mean Weight Females: 135.86 
- Standard Deviation Weight Females: 19.02 

#### 5. Is variability greater in height or in weight ? 
In all instances, the standard deviation is larger for wieght than height- this shows that there is greater variability in weight, regardless of gender. 

#### 6. What can the variability tell you about which variables are more easily influenced ?
Higher variability means that the values deviate more widely from the mean. This means that outside factors can have a greater impact on the variable. Eg. Weight is more variable than height, weight can easily be impacted by things like diet. Lower variability means it more consistent through individuals. Lower variability suggests that a variable is more consistent throughtout a population. Eg. In our example, height is more consistent between people of that population than weight is- and less likely to be impacted by outside factors than weight. 

#### 7. Height and weight have increased over the years. So if you were to go back 50 years, which measurement would you guess has changed most, compared to today's numbers, height or weight (remember you are comparing height and weight which use different units, so what is the best metric to use here ?

Since we see that weight has a higher variability, I would guess that weight has changed more in the last 50 years compared to height. The best metric to use to verify this would be percent change in comparing the changes in means between the two. See what percent each mean has increased by  (for both height and weight). 